### DISCUSSION TITLE
Unable to sent / receive long messages with Heltec Wireless Stick Lite (V3)

### DISCUSSION THREAD

**Discussion by FlavorFx** [jgromes/RadioLib#847] [Q&A]
I have two Heltec Wireless Stick Lite (V3), SX1262 board and took these sample programs to sent and receive messages:
- https://github.com/jgromes/RadioLib/blob/master/examples/SX126x/SX126x_Transmit_Blocking/SX126x_Transmit_Blocking.ino
- https://github.com/jgromes/RadioLib/blob/master/examples/SX126x/SX126x_Receive_Blocking/SX126x_Receive_Blocking.ino

The communication between these boards work fine but when I extend the message size from 39 to 53 characters, the communication stopped working. Same behavior with another board with SX1276. I have no glue, why. With sandeepmistry's lora lib the Sx1276 communication works fine but it does not support SX1262. I compared the default setting of each lib but no success.

Any hint is appreciated.

**Comment by jgromes** [comment]
> when I extend the message size from 39 to 53 characters, the communication stopped working

What exactly do you mean by that? Does the transmitter/receiver return some error code? And does the issue persist when using interrupt receive?

**Reply by FlavorFx** [reply]
When I enlarge the msg size, the receiver does not show any received msg anymore. But there is no error message in the log. I haven't tried interrupt received.

**Reply by jgromes** [reply]
So in that case I'm guessing the receiver just times out. I would suggest using the interrupt receive instead, since blocking always waits for 100 LoRa symbols, so the longer the message, the less of a chance you have to receive the packet. The interrupt receive just listens forever.

**Reply by FlavorFx** [reply]
Thanks for your suggestion. The interrupt receive method works fine now to get long messages. Unfortunately, I', still puzzled. 

The communication with radiolib between my two SX1262 works fine. But these two do not communicate with my SX1276 Gateway (lora.lib). 

And another SX1276 doesn't communicate with these two SX1262 nor with my gateway. The communication to the gateway works properly with lora.lib. I ensured that the Sync Word is set to the same value (0x12). 

There are no error messages in the log. Do you have any idea or hint?

**Reply by jgromes** [reply]
It's impossible to know without seeing the code/configuration on both sides. When only using RadioLib, SX127x will be able to talk to SX126x with the default settings, but I have no idea what is the configuration of your other library. Sync word is not the only possible configuration, you can have different header configuration, IQ inversion etc.

**Reply by FlavorFx** [reply]
Pls have a look to these files: https://gist.github.com/FlavorFx/04349b7e4622536c561d3d8167277d80.
"main with loralib.cpp" works fine with my gateway. "SX126x_Tx" and "SX1262Rx_IRQ" works fine each other, but "SX1276_Tx" does not reach my gateway nor my receiver.

**Reply by jgromes** [reply]
You don't seem to be configuring anything apart from the sync word. What is the default frequency of your non-RadioLib receiver and the gateway? What are its spreading factors, bandwidths, coding rates, ...?

This page has the default configuration for RadioLib: https://github.com/jgromes/RadioLib/wiki/Default-configuration

**Reply by FlavorFx** [reply]
I checked the defaults and noticed a difference of the sync word. All parameters looks equal. All devices works with 868MHz. I'll double check the parameter again.

**Reply by FlavorFx** [reply]
Finally, it works. I took care to use the same settings of lora configuration. Thanks a lot for you support.
